cypress/integration/ResourcesForResearchers/index.js

```diff
@@ -0,0 +1,264 @@
+  cy.get('.searchbar__container input').type('{enter}{enter}');
+  cy.wait(2000);
+});
+
+When('user clicks on {string} button', (button) => {
+  cy.get('.r4r-DEFAULT.results__search-container > .r4r-DEFAULT.results__filter-button').click();
+});
+
+And('the search result is displayed', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+});
+
+And('the search result title is {string} with href {string}', (title, href) => {
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href', href).should('have.text', title);
+});
+
+And('the search result description is displayed', () => {
+  cy.get('[aria-label="search results"] > article>p').first().should('be.visible');
+});
+
+And('the message is displayed as {string}', (message) => {
+  cy.get('.results__noresults').should('contain.text', message);
+});
+
+And("{string} is a link with href {string}", (linkText, linkHref) => {
+  cy.get('.results__noresults > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+Then('the resource title is displayed as {string}', (resourceTitle) => {
+  cy.get('.r4r-resource__header h1').should('contain.text', resourceTitle);
+});
+
+And("{string} is displayed with href {string}", (linkText, linkHref) => {
+  cy.get('.r4r-DEFAULT.resource__home > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+// And('the {string} is displayed', () => {
+//   cy.get('[aria-label="Resource description"]>p').should('be.visible');
+// });
+
+And('Visit Resource link is displayed', () => {
+  cy.get('.resource__link--external').should('be.visible');
+});
+
+And('search box is displayed', () => {
+  cy.get('.searchbar__container input').should('be.visible');
+});
+
+And('{string} section is displayed', (section) => {
+  cy.get('.resource__nav h2').should('be.visible');
+});
+
+And('{string} field is displayed', (fieldName) => {
+  cy.document().then(doc => {
+    expect(doc.querySelector(`[aria-label="${fieldName}"]`)).to.exist;
+  })
+});
+
+And('Contact Information field is displayed', (fieldName) => {
+  cy.document().then(doc => {
+    const art = doc.querySelectorAll(`.r4r-DEFAULT.resource__main > article`);
+    expect(art[2]).to.exist;
+  })
+
+});
+
+And('the Resource not found message is displayed as {string} and {string}', (message1, message2) => {
+  cy.document().then(doc => {
+
+    const msg = doc.querySelectorAll(`.r4r-DEFAULT.r4r-error > p`);
+    expect(msg[0].innerText).to.eq(message1);
+    expect(msg[1].innerText).to.eq(message2);
+
+  })
+});
+
+And("the {string} is displayed with href {string}", (linkText, linkHref) => {
+  cy.get('.r4r-container.row a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And('the {string} box displays the filter {string} as checked', (section, area) => {
+  cy.document().then(doc => {
+    const box = doc.querySelectorAll(`.r4r-DEFAULT.facet__box >h4`);
```

Comment:
no need to use cy.document

10 lines of code becomes 1 : 
`cy.get('h4').contains(section).parent().find(`input[value="${area}"]`).should('be.checked')`

Reply:
fixed it, thanks